src/filesystem/implementation.js

```diff
@@ -1678,6 +1678,25 @@ function mkdir(fs, context, path, mode, callback) {
   make_directory(context, path, callback);
 }
 
+function mkdtemp(fs, context, prefix, options, callback) { 
+  callback = arguments[arguments.length - 1];
+  let shared = require('../shared.js');
```

Comment:
Let's move this to the top of the file vs. doing it every time we call this function.  Also, use `var`, even though `let` is correct, as we haven't converted over to `const` and `let` yet.